core/CMakeLists.txt

```diff
@@ -482,25 +482,6 @@ endif()
 
 set_target_properties(ZXing PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}")
 
-if (APPLE AND BUILD_APPLE_FRAMEWORK)
-    set_target_properties(ZXing PROPERTIES
-        FRAMEWORK TRUE
-        FRAMEWORK_VERSION "C"
-        XCODE_ATTRIBUTE_DEFINES_MODULE YES
-        XCODE_ATTRIBUTE_BUILD_LIBRARY_FOR_DISTRIBUTION YES
-        XCODE_ATTRIBUTE_MODULEMAP_FILE "wrappers/ios/Sources/Wrapper/module.modulemap"
-        XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
-        MACOSX_FRAMEWORK_IDENTIFIER "com.zxing-cpp.ios"
-        MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${PROJECT_VERSION}
-        MACOSX_FRAMEWORK_BUNDLE_VERSION ${PROJECT_VERSION}
-        CMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH NO
-        #MACOSX_FRAMEWORK_INFO_PLIST Info.plist
-        PUBLIC_HEADER "${PUBLIC_HEADERS}"
-        XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "iPhone Developer"
-        XCODE_ATTRIBUTE_ENABLE_BITCODE "NO"
-    )
-endif()
-
```

Comment:
This is obsolete! But I am glad you are asking these questions:

These additional bits were necessary to craft Frameworks merged into an XCFramework (containing the static built library - actually some renamed `*.a` file).

Framework is a library (either dynamically or statically linked) for one particular platform. It contains a bit of Apple-related meta information, that's all. XCFramework is a collection of multiple Frameworks to work (depends on how you built it, but that's the idea) on every platform (w.r.t. Apple) - an universal framework so to speak.

An `.a` (static) or in case of Apple `.dylib` (dynamic) file gets generated by the Swift Package Manager (which you can easily use) by `swift build` in the root folder (depending where the `Package.swift` file is located).

So there is no real downside in this approach, you can still build the file if you just need the library without any additional foo.

Reply:
Ah forgot to mention, as we can now include `zxing-cpp` as a direct dependency with the Xcode dependency management, Xcode takes care on building and including `zxing-cpp` in the app / project.

Reply:
Just to clarify that: you can do both, use it in your app or just build solely the library.